<launch>
	<!-- Common parameters -->
	<arg name="camera_name" default="camera" />
	<arg name="frame_rate" default="30" />

	<arg name="left_camera_serial" default="14150439" />
	<arg name="left_camera_calibrated" default="0" />

	<arg name="right_camera_serial" default="14150441" />
	<arg name="right_camera_calibrated" default="0" />

	<group ns="$(arg camera_name)" >
		<group ns="left" >
			<node pkg="nodelet" type="nodelet" name="camera_nodelet_manager" args="manager" output="screen"/>

			<!-- Both cameras are not loaded into one nodelet manager to avoid the namespacing issue. -->
			<node pkg="nodelet" type="nodelet" name="camera_nodelet"  output="screen"
				  args="load pointgrey_camera_driver/PointGreyCameraNodelet camera_nodelet_manager" >
				<param name="frame_id" value="camera_left" />
				<param name="serial" value="$(arg left_camera_serial)" />

				<!-- When unspecified, the driver will use the default framerate as given by the
                     camera itself. Use this parameter to override that value for cameras capable of
                     other framerates. -->
				<param name="frame_rate" value="$(arg frame_rate)" />

				<!-- Use the camera_calibration package to create this file -->
				<param name="camera_info_url" if="$(arg left_camera_calibrated)"
					   value="file://$(env HOME)/.ros/camera_info/$(arg left_camera_serial).yaml" />
			</node>

			<node pkg="nodelet" type="nodelet" name="image_proc_debayer"  output="screen"
				  args="load image_proc/debayer camera_nodelet_manager">
			</node>


	<!--		<node pkg="nodelet" type="nodelet" name="extrinsic_nodelet"  output="screen"
			args="load teleop_vision/ExtrinsicArucoNodelet camera_nodelet_manager">
				<rosparam command="load" file="$(find teleop_vision)/launch/params_aruco_board.yaml" />
				<param name="cam_intrinsic_calibration_file_path" value="$(find teleop_vision)/resources/flea_left_intrinsic.xml" />
				<param name="image_transport_namespace" value="/$(arg camera_name)left/image_color/"/>
				<param name="board_to_cam_pose_topic_name" value="board_to_camera"/>
				<param name="draw_axes" value="true"/>
				<param name="image_transport" value="resized"/>
			</node>
			<param name="/$(arg camera_name)/left/image_color/resized/downscaling_factor" value="4"/>

			<node pkg="teleop_vision" type="extrinsic_calib_aruco" name="extrinsic_aruco" output="screen" >
				<rosparam command="load" file="$(find teleop_vision)/launch/params_aruco_board.yaml" />
				<param name="cam_intrinsic_calibration_file_path" value="$(find teleop_vision)/resources/flea_left_intrinsic.xml" />
				<param name="image_transport_namespace" value="/$(arg camera_name)/left/image_color/"/>
				<param name="board_to_cam_pose_topic_name" value="board_to_camera"/>
				<param name="draw_axes" value="true"/>
				<param name="frequency" value="30"/>
				<param name="image_transport" value="compressed"/>
			</node>
-->			<!--<param name="/$(arg camera_name)/left/image_color/resized/downscaling_factor" value="4"/>-->

		</group>



	</group>
</launch>
